### PR TITLE
Stop throwing errors.

### PR DESCRIPTION
**Pull request by MancunianSam**
It looks like that some of our error handling is causing the javascript
to throw an error when there is a temporary failure of the upload which
is corrected by the SDK so we shouldn't be throwing an error.

We're using thrown errors for normal validation which is not what
they're for really, they should be for exceptional failures.
 The reason we did this originally is that we would end up with a load of
branching if/else for the errors if we didn't throw Errors for validation.

Now that throwing errors is causing problems, we're going to have to
live with the branching if/else statements until Typescript sorts out a
decent function style of error handling like Scala's Either.

### CODE REVIEW

**Comment by MancunianSam** [comment]
I've tried monet.js. The problem is most things return promises which can be converted to `Promise<Either<Error,T>>` which can be `await`ed to give `Either<Error,T>`
This then needs to be `flatMap`ped with another Promise which needs to be `await`ed again but it can't be because the type signature of flatMap doesn't allow a promise return type.

I also tried https://github.com/fluture-js/Fluture This is better but there's quite a large number of changes, some examples are here https://github.com/nationalarchives/tdr-transfer-frontend/blob/use-either-for-errors/npm/src/clientfilemetadataupload/index.ts.
It seems like a lot of changes to fix something we're not even sure is a problem so I think the if statements are the lesser of two evils.

**Review by TomJKing — COMMENTED** [review]
Looks good. Just one very minor comment, feel free to ignore

File: npm/src/auth/index.ts [hunk 1]
```diff
@@ -60,19 +67,28 @@ export const scheduleTokenRefresh: (
...
+        return new Error(err)
+      })
     }
-    await keycloak.updateToken(tokenMinValidityInSecs)
   }
   if (keycloak.token) {
     return keycloak.token
   } else {
+    //We shouldn't normally throw Errors but this is exceptional
     throw "Something really odd has happened, Keycloak is down or something"
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply npm/src/auth/index.ts]: The url is available on the `KeycloakInstance` object so I've included that and reworded the error message slightly.

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**